nncf/torch/model_transformer.py

```diff
@@ -97,6 +102,42 @@ def _apply_insertion_transformations(model: NNCFNetwork, transformations: List[P
 
         return model
 
+    @staticmethod
+    def _apply_quantizer_insertion_transformations(
+        model: NNCFNetwork, transformations: List[PTQuantizerInsertionCommand]
+    ) -> NNCFNetwork:
+        """
+        Applies quantizer insertion transformations on the model.
+
+        :param model: Model to apply transformations.
+        :param transformations: List of the OVQuantizerInsertionCommand transformations.
+        :return: Model with inserted FakeQuantize nodes.
+        """
+        node_to_op_address_mapping = model.nncf.get_node_to_op_address_mapping()
+
+        model.nncf.register_compression_module_type(ExtraCompressionModuleType.EXTERNAL_QUANTIZER)
+
+        for transformation_command in transformations:
+            target_point: PTTargetPoint = transformation_command.target_point
+            target_node_name = target_point.target_node_name
```

Comment:
Do you have any thoughts on how we can incorporate shared quantizers logic into here?

Reply:
I think that the shared quantizer handling is already there in the NNCF PT [code](https://github.com/openvinotoolkit/nncf/blob/631a7b8f2990a80b9b9473f999651b1808a7e7d4/nncf/torch/quantization/algo.py#L955C5-L966), you are just continuously deciding not to use it and roll your own code instead.